nautobot/core/jobs/bulk_actions.py

```diff
@@ -0,0 +1,80 @@
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import (
+    PermissionDenied,
+)
+from django.db.models import ProtectedError
+
+from nautobot.core.utils.lookup import get_filterset_for_model, get_form_for_model
+from nautobot.extras.jobs import (
+    BooleanVar,
+    Job,
+    JSONVar,
+    ObjectVar,
+    RunJobTaskFailed,
+)
+from nautobot.extras.utils import bulk_delete_with_bulk_change_logging
+
+
+class BulkDeleteObjects(Job):
+    """
+    System job to bulk delete objs`.
+    """
+
+    content_type = ObjectVar(
+        model=ContentType,
+        description="Type of objects to import",
+    )
+    pk_list = JSONVar(description="PK List of objects to delete", required=False)
+    delete_all = BooleanVar(description="Bulk Delete all object / all filtered objects", required=False)
+    filter_query_params = JSONVar(label="Filter Query Params", required=False)
+
+    class Meta:
+        name = "Bulk Delete Objects"
+        description = "Bulk delete objects."
+        has_sensitive_variables = False
+        soft_time_limit = 1800
+        time_limit = 2000
+
+    def run(self, *, content_type, pk_list=None, delete_all=False, filter_query_params=None):
+        if not self.user.has_perm(f"{content_type.app_label}.delete_{content_type.model}"):
+            self.logger.error('User "%s" does not have permission to delete %s objects', self.user, content_type.model)
+            raise PermissionDenied("User does not have delete permissions on the requested content-type")
+
+        model = content_type.model_class()
+        if model is None:
+            self.logger.error(
+                'Could not find the "%s.%s" data model. Perhaps an app is uninstalled?',
+                content_type.app_label,
+                content_type.model,
+            )
+            raise RunJobTaskFailed("Model not found")
+
```

Comment:
Can we add a check to enforce `pk_list` and `delete_all` are mutually exclusive? (And a test case verifying this behavior?)

Maybe `pk_list` and `filter_query_params` mutually exclusive as well?